src/jdk.attach/share/classes/sun/tools/attach/HotSpotVirtualMachine.java

```diff
@@ -45,15 +45,14 @@
  * The HotSpot implementation of com.sun.tools.attach.VirtualMachine.
  */
 
-@SuppressWarnings("removal")
 public abstract class HotSpotVirtualMachine extends VirtualMachine {
 
-    private static final long CURRENT_PID;
+    @SuppressWarnings("removal")
+    private static final long CURRENT_PID = AccessController.doPrivileged(
+            (PrivilegedAction<ProcessHandle>) ProcessHandle::current).pid();
+
```

Comment:
The original code separated out the declaration of the PrivilegedAction to avoid this cast. If you move the code from the original static initializer into a static method that it called from initializer then it might provide you with a cleaner way to refactor this. There are several other places in this patch that could do with similar cleanup.

Reply:
This cast is only to tell the compiler which overloaded method to call, and I don't think there will be a real cast at runtime. It might look a little ugly but extracting it into a variable declaration/definition plus a new `initStatic` method seems not worth doing, IMHO.

Reply:
Why not simply declare a local variable in the static initializer below?

```
    private static final long CURRENT_PID;
    private static final boolean ALLOW_ATTACH_SELF;
    static {
        PrivilegedAction<ProcessHandle> pa = ProcessHandle::current;
        @SuppressWarnings("removal")
        long pid =AccessController.doPrivileged(pa).pid();
        CURRENT_PID = pid;
        String s = VM.getSavedProperty("jdk.attach.allowAttachSelf");
        ALLOW_ATTACH_SELF = "".equals(s) || Boolean.parseBoolean(s);
    }
```

Reply:
I've just pushed a commit with a different fix:
```
    private static final long CURRENT_PID = pid();

    @SuppressWarnings("removal")
    private static long pid() {
        PrivilegedAction<ProcessHandle> pa = () -> ProcessHandle.current();
        return AccessController.doPrivileged(pa).pid();
    }
```